.github/workflows/Build.yaml

```diff
@@ -106,7 +69,19 @@ jobs:
       # Run local tests after screenshot tests to avoid wrong UP-TO-DATE. TODO: Ignore screenshots.
       - name: Run local tests
         if: always()
-        run: ./gradlew testDemoDebug testProdDebug
+        run: ./gradlew testDemoDebug testProdDebug :lint:test
+
+      - name: Check spotless
+        run: ./gradlew spotlessCheck --init-script gradle/init.gradle.kts --no-configuration-cache
```

Comment:
nit: could we avoid checking for spotless this late in the job?
Formatting issues happen very frequently, and checking for them after compiling and testing the codebase wastes precious time (for devs and the CI).

This also means, that removing `needs: build` from `androidTest` now will run entirely, even if formatting issues happen.

Reply:
Ah, I meant to do that, thanks.

> This also means, that removing needs: build from androidTest now will run entirely, even if formatting issues happen.

Yeah this is fine, as long as one job fails.

Reply:
> Yeah this is fine, as long as one job fails.

I'm really uncomfortable with this. This will waste ~40 minutes of CI every time this happens.